providers/implementations/ciphers/cipher_tdes_common.c

```diff
@@ -92,7 +109,13 @@ static int tdes_init(void *vctx, const unsigned char *key, size_t keylen,
             return 0;
         ctx->key_set = 1;
     }
-    return ossl_cipher_generic_set_ctx_params(ctx, params);
+    if (!ossl_tdes_set_ctx_params(ctx, params))
+        return 0;
+#ifdef FIPS_MODULE
```

Comment:
I would really like others opinions about where this check should happen..
Because of the stupid way the Init API works, this could get triggered multiple times when multiple init calls happen.

I could leave it to check in the update but that seems late.

Reply:
e.g. If we wanted cmac_prov to have an indicator related to the cipher encrypt, cmac_prov.c calls CMAC_init() which does the encrypt_init internally so I would have to have another function if we wanted to allow CMAC_init to pass params to its init. Doing it during the update would be easier.

Reply:
You could just call OSSL_FIPS_IND_ON_UNAPPROVED in ossl_tdes_einit, but I'm not sure how much that saves you

Reply:
> e.g. If we wanted cmac_prov to have an indicator related to the cipher encrypt

We definitely want CMAC to include the validity the the cipher used.  I don't think this is captured in the epic.